### PR TITLE
consolidate enums into DSharpPlus.Entities; enforce consistent naming

### PR DESCRIPTION
**Pull request by akiraveliara**
# This is a massive breaking change.

As of now, all entity-related enums should be in DSharpPlus/Entities, in the namespace DSharpPlus.Entities and Discord- prefixed. This also fixes a few cases where entities weren't Discord- prefixed (ahem, velvet.)

Previously, some enums were in DSharpPlus, some in DSharpPlus.Enums, some in DSharpPlus.Entities unprefixed, some in DSharpPlus.Entities prefixed. as massive as this break is, I think it's worth it to address this and to give users one unconditionally correct way to address our enums.

### CODE REVIEW

File: DSharpPlus/Entities/AuditLogs/AuditLogParser.cs [hunk 1]
```diff
@@ -100,11 +96,6 @@ internal static class AuditLogParser
         IOrderedEnumerable<AuditLogAction>? auditLogActions = auditLog.Entries.OrderByDescending(xa => xa.Id);
         foreach (AuditLogAction? auditLogAction in auditLogActions)
         {
-            if (cancellationToken.IsCancellationRequested)
```
Comment (Plerx2493) [inline DSharpPlus/Entities/AuditLogs/AuditLogParser.cs:103]: please rebase

File: DSharpPlus/GatewayCompressionLevel.cs
Comment (Plerx2493) [inline DSharpPlus/GatewayCompressionLevel.cs:1]: Shouldnt this be `DiscordGatewayCompressionLevel`?

File: DSharpPlus/TokenType.cs
Comment (Plerx2493) [inline DSharpPlus/TokenType.cs:1]: Shouldnt this be `DiscordTokenType`?

**Review by Plerx2493 — CHANGES_REQUESTED** [review]

**Comment by Plerx2493** [comment]
oh, i missed Timestampforma, its also missing the discord prefix

File: DSharpPlus/Entities/AuditLogs/AuditLogParser.cs (hunk 1 above)
Comment (akiraveliara) [reply DSharpPlus/Entities/AuditLogs/AuditLogParser.cs:103]: despair

File: DSharpPlus/GatewayCompressionLevel.cs
Comment (akiraveliara) [reply DSharpPlus/GatewayCompressionLevel.cs:1]: no, it's a library setting rather than an entity enum

File: DSharpPlus/TokenType.cs
Comment (akiraveliara) [reply DSharpPlus/TokenType.cs:1]: same thing here

**Review by Plerx2493 — APPROVED** [review]

**Pull request merged**